utils/symbol-rawelf.c

```diff
@@ -13,6 +13,96 @@
 #include "utils/utils.h"
 #include "utils/symbol-rawelf.h"
 
+/*
+ *  ELF File Header validation logic.
+ */
+int elf_validation(struct uftrace_elf_data *elf)
+{
+	Elf_Ehdr *ehdr;
+	int eclass, data, version;
+	int res = 0;
+	unsigned long mapped_saddr, mapped_eaddr, mapped_size, addr;
+
+	ehdr = &elf->ehdr;
+
+	// validate ELF Magic.
+	if (memcmp(ehdr, ELFMAG, SELFMAG)) {
+		pr_dbg("ELF Signature not matched\n");
+		res = -1;
+	}
+
+	// validate some field of elf header.
+	eclass = (int) ((unsigned char *) ehdr)[EI_CLASS];
+	data = (int) ((unsigned char *) ehdr)[EI_DATA];
+	version = (int) ((unsigned char *) ehdr)[EI_VERSION];
+
+	if (!(eclass > ELFCLASSNONE && eclass < ELFCLASSNUM)) {
+		pr_dbg("eclass out of range. it must be inside range "\
+				"between %d and %d, but Yours is [%d]\n",
+				ELFCLASSNONE, ELFCLASSNUM, eclass);
```

Comment:
generally I don't like multi-line strings especially in debug level 1.  Also I think this verbose info is not very useful to users.  What about just like below?

```
   pr_dbg("invalid ELF class: %d\n", eclass);
```

Reply:
Yes, but this message appears to be valuable to the developer. 
So, how about leave a message and just raise the level to 2 or 3 instead?

Reply:
That's why I printed the class number.  I guess relavant developers know what the value means and can check it's invalid easily.

Reply:
maybe, but I think it cost a little bit.